marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -34,15 +39,16 @@ object Api {
     }
   }
 
-  def apply(client: Client, host: String, uriPrefix: String): Api =
-    new AppIdApi(SetHost(host).andThen(client), s"$uriPrefix/$versionString")
+  def apply(client: Client, host: String, uriPrefix: String, auth: Option[Auth]): Api =
+    new AppIdApi(SetHost(host).andThen(client), s"$uriPrefix/$versionString", auth)
 
   private[v2] def rspToApps(rsp: http.Response): Future[Api.AppIds] =
     rsp.status match {
       case http.Status.Ok =>
         val apps = readJson[AppsRsp](rsp.content).map(_.toApps)
         Future.const(apps)
-
+      case http.Status.Unauthorized =>
+        Future.exception(UnauthorizedResponse(rsp))
```

Comment:
we probably want to log here (and probably also in the UnexpectedResponse case as well...)